packages/jaeger-ui/.depcheckrc.json

```diff
@@ -0,0 +1,12 @@
+{
+  "ignores": [
+    "@babel/preset-env",
```

Comment:
In that case, we don't need this package altogether.

But, creating a custom script to analyze the whole codebase, and checking for deps in the codebase against the deps inside package.json is what I wouldn't advocate for, though it can be done, but would be more like reinventing the wheel.

In case we aren't sure of the list (given to us by depcheck), we can go ahead with another dep checker, i.e., [npm-check](https://www.npmjs.com/package/npm-check).
I manually checked with it also, and the list is same as below:
<details>
<summary>Output</summary>

```
tar                                          😕  NOTUSED?  Still using tar?
                                                          Depcheck did not find code similar to require('tar') or import from 'tar'.
                                                          Check your code before removing as depcheck isn't able to foresee all ways dependencies can be used.
                                                          Use rc file options to remove unused check, but still monitor for outdated version:
                                                              .npmcheckrc {"depcheck": {"ignoreMatches": ["tar"]}}
                                                          Use --skip-unused to skip this check.
                                                          To remove this package: yarn remove tar 

@babel/preset-env                            😕  NOTUSED?  Still using @babel/preset-env?
                                                          Depcheck did not find code similar to require('@babel/preset-env') or import from '@babel/preset-env'.
                                                          Check your code before removing as depcheck isn't able to foresee all ways dependencies can be used.
                                                          Use rc file options to remove unused check, but still monitor for outdated version:
                                                              .npmcheckrc {"depcheck": {"ignoreMatches": ["@babel/preset-env"]}}
                                                          Use --skip-unused to skip this check.
                                                          To remove this package: yarn remove @babel/preset-env --dev

@babel/preset-react                          😕  NOTUSED?  Still using @babel/preset-react?
                                                          Depcheck did not find code similar to require('@babel/preset-react') or import from '@babel/preset-react'.
                                                          Check your code before removing as depcheck isn't able to foresee all ways dependencies can be used.
                                                          Use rc file options to remove unused check, but still monitor for outdated version:
                                                              .npmcheckrc {"depcheck": {"ignoreMatches": ["@babel/preset-react"]}}
                                                          Use --skip-unused to skip this check.
                                                          To remove this package: yarn remove @babel/preset-react --dev

@babel/preset-typescript                     😕  NOTUSED?  Still using @babel/preset-typescript?
                                                          Depcheck did not find code similar to require('@babel/preset-typescript') or import from '@babel/preset-typescript'.
                                                          Check your code before removing as depcheck isn't able to foresee all ways dependencies can be used.
                                                          Use rc file options to remove unused check, but still monitor for outdated version:
                                                              .npmcheckrc {"depcheck": {"ignoreMatches": ["@babel/preset-typescript"]}}
                                                          Use --skip-unused to skip this check.
                                                          To remove this package: yarn remove @babel/preset-typescript --dev

@svgr/babel-plugin-transform-svg-component   😟  MISSING!  Not installed.
                                             😕  NOTUSED?  Still using @svgr/babel-plugin-transform-svg-component?
                                                          Depcheck did not find code similar to require('@svgr/babel-plugin-transform-svg-component') or import from '@svgr/babel-plugin-transform-svg-component'.
                                                          Check your code before removing as depcheck isn't able to foresee all ways dependencies can be used.
                                                          Use rc file options to remove unused check, but still monitor for outdated version:
                                                              .npmcheckrc {"depcheck": {"ignoreMatches": ["@svgr/babel-plugin-transform-svg-component"]}}
                                                          Use --skip-unused to skip this check.
                                                          To remove this package: yarn remove @svgr/babel-plugin-transform-svg-component --dev

@svgr/babel-preset                           😟  MISSING!  Not installed.
                                             😕  NOTUSED?  Still using @svgr/babel-preset?
                                                          Depcheck did not find code similar to require('@svgr/babel-preset') or import from '@svgr/babel-preset'.
                                                          Check your code before removing as depcheck isn't able to foresee all ways dependencies can be used.
                                                          Use rc file options to remove unused check, but still monitor for outdated version:
                                                              .npmcheckrc {"depcheck": {"ignoreMatches": ["@svgr/babel-preset"]}}
                                                          Use --skip-unused to skip this check.
                                                          To remove this package: yarn remove @svgr/babel-preset --dev

@testing-library/jest-dom                    😎  MAJOR UP  Major update available. https://github.com/testing-library/jest-dom#readme
                                                          yarn add @testing-library/jest-dom@6.0.0 --dev to go from 5.17.0 to 6.0.0

@types/match-sorter                          😎  MAJOR UP  Major update available.
                                                          yarn add @types/match-sorter@6.0.0 --dev to go from 2.3.0 to 6.0.0

@types/react                                 😎  MAJOR UP  Major update available. https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/react
                                                          yarn add @types/react@18.2.20 --dev to go from 16.8.7 to 18.2.20

@types/react-dom                             😎  PATCH UP  Patch update available. https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/react-dom
                                                          yarn add @types/react-dom@18.2.7 --dev to go from 18.2.5 to 18.2.7

@types/react-router-redux                    😍  UPDATE!   Your local install is out of date. https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/react-router-redux
                                                          yarn add @types/react-router-redux@5.0.22 --dev to go from 5.0.21 to 5.0.22

jest                                         😎  MAJOR UP  Major update available. https://jestjs.io/
                                                          yarn add jest@29.6.2 --dev to go from 28.1.3 to 29.6.2

jest-environment-jsdom                       😍  UPDATE!   Your local install is out of date. https://github.com/facebook/jest#readme
                                                          yarn add jest-environment-jsdom@29.6.2 --dev to go from 29.6.1 to 29.6.2
                                             😕  NOTUSED?  Still using jest-environment-jsdom?
                                                          Depcheck did not find code similar to require('jest-environment-jsdom') or import from 'jest-environment-jsdom'.
                                                          Check your code before removing as depcheck isn't able to foresee all ways dependencies can be used.
                                                          Use rc file options to remove unused check, but still monitor for outdated version:
                                                              .npmcheckrc {"depcheck": {"ignoreMatches": ["jest-environment-jsdom"]}}
                                                          Use --skip-unused to skip this check.
                                                          To remove this package: yarn remove jest-environment-jsdom --dev

less                                         😎  MAJOR UP  Major update available. http://lesscss.org
                                                          yarn add less@4.2.0 --dev to go from 3.13.1 to 4.2.0

sinon                                        😎  MAJOR UP  Major update available. https://sinonjs.org/
                                                          yarn add sinon@15.2.0 --dev to go from 7.3.2 to 15.2.0

terser                                       😍  UPDATE!   Your local install is out of date. https://terser.org
                                                          yarn add terser@5.19.2 --dev to go from 5.19.0 to 5.19.2

react-is                                     😟  PKG ERR!  Not in the package.json. Found in: /src/utils/ReactShallowRenderer.test.js
                                             😎  MAJOR UP  Major update available. https://reactjs.org/
                                                          yarn add react-is@18.2.0  to go from 16.13.1 to 18.2.0

react-select                                 😟  PKG ERR!  Not in the package.json. Found in: /src/components/common/VirtSelect.tsx
                                             😎  MAJOR UP  Major update available. https://github.com/JedWatson/react-select/tree/master#readme
                                                          yarn add react-select@5.7.4  to go from 1.3.0 to 5.7.4

react-virtualized                            😟  PKG ERR!  Not in the package.json. Found in: /src/components/common/VirtSelect.tsx
                                             😎  MINOR UP  Minor update available. https://github.com/bvaughn/react-virtualized
                                                          yarn add react-virtualized@9.22.5  to go from 9.21.0 to 9.22.5

Use npm-check -u for interactive update.
```

</details>

The good thing about this tool is, it didn't gave a false +ve on @types deps, but downside is, it is very slow.

Reply:
> But, creating a custom script to analyze the whole codebase

This is now that I was suggesting. Let's say there is a list of plugins mentioned in `.babelrc` or something similar. If depcheck does not verify that file (a reasonable behavior, it cannot know about all possible libraries out there and their bespoke configuration), we can still write a smalls script that will read `.babelrc` or simiar and output a temporary `.depcheckrc.json`. This way everything stays consistent - if we remove a plugin from babel, the CI will fail if we it's not removed from dependencies as well.

Reply:
I got the scenario here, but the issue is with what would we compare those list of dependencies with, which we parse from the files like `.babelrc`

For example:

`{A, B}` are parsed from `.babelrc`, `{C, D}` is parsed from `vite.config.js`.

Now, we have `package.json` with `{A, B, C, D, E, F, . . . . n}`.
Here, we would compare the parsed result (`{A, B, C, D}`) with `A, B, C, D`, but what about `E, F, . . . n`
We don't know if they also contain any unused dep, since that part is managed by Depcheck.

There are some other dependencies that are not explicitly required but need to be installed.
For example: `jest-environment-jsdom`. 
We cannot parse it from any config file, though it is needed if we use the following configuration in our `package.json`
```
"testEnvironment": "jsdom"
```
which is the case for us.

We can also configure our parser to add the dep `jest-environment-jsdom` to ignore list in case it encounters `"testEnvironment": jsdom`

Reply:
> what would we compare those list of dependencies with

Why do we need to compare? We can parse those known config files, extract plugin names, and write them out into the ignore section of .depcheckrc, then just run depcheck